examples/gno.land/r/demo/keystore/keystore.gno

```diff
@@ -0,0 +1,189 @@
+package keystore
+
+import (
+	"std"
+	"strings"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+var data avl.Tree
+
+var (
+	BaseURL               = "/r/demo/keystore"
+	StatusOK              = "ok"
+	StatusNoUser          = "user not found"
+	StatusNotFound        = "key not found"
+	StatusNoWriteAccess   = "no write access"
+	StatusCouldNotExecute = "could not execute"
+)
+
+func init() {
+	data = avl.Tree{} // user -> avl.Tree
+}
+
+// KeyStore stores the owner-specific avl.Tree
+type KeyStore struct {
+	Owner std.Address
+	Data  avl.Tree
+}
+
+// Set will set a value to a key
+// requires write-access (original caller must be caller)
+func Set(k, v string) string {
+	origOwner := std.GetOrigCaller()
+	return set(origOwner.String(), k, v)
+}
+
+// set (private) will set a key to value
+// requires write-access (original caller must be caller)
+func set(owner, k, v string) string {
+	origOwner := std.GetOrigCaller()
+	if origOwner.String() != owner {
+		return StatusNoWriteAccess
+	}
+	keystoreInterface, exists := data.Get(owner)
+	if !exists {
+		data.Set(owner, &KeyStore{
+			Owner: origOwner,
+			Data:  avl.Tree{},
+		})
+		keystoreInterface, _ = data.Get(owner)
+	}
+	keystore := keystoreInterface.(*KeyStore)
+	keystore.Data.Set(k, v)
+	return StatusOK
```

Comment:
Why are you setting the keystore, and then fetching it again from the map (avl)?

Why not simply define a `var keystore KeyStore` outside the condition, and if it doesn't exist, assign it in the branch?

Reply:
You are totally right, thanks for that. Its fixed in 61079bb8